src/viewProviders/collapsing.ts

```diff
@@ -0,0 +1,39 @@
+import { TreeItem, TreeItemCollapsibleState } from "vscode";
+import { ISearchable } from "../models/resource";
+import { matchesOrHasMatchingChild } from "./search";
+
+/**
+ * Adjust the {@link TreeItemCollapsibleState TreeItemCollapsibleState} of a tree item that
+ * was returned from a tree data provider's `getChildren()` while a search string was applied.
+ */
+export function updateCollapsibleStateFromSearch(
+  element: ISearchable,
+  treeItem: TreeItem,
+  searchStr: string,
+): TreeItem {
+  if (!searchStr) {
+    // return the tree item as-is if there's no search string or it's empty
+    return treeItem;
+  }
+
+  const origCollapsibleState = treeItem.collapsibleState;
+
+  if (element.children?.some((child) => matchesOrHasMatchingChild(child, searchStr))) {
+    // has children that match the search
+    treeItem.collapsibleState = TreeItemCollapsibleState.Expanded;
+  } else if (element.children?.length) {
+    // has children, but none of them match
+    treeItem.collapsibleState = TreeItemCollapsibleState.Collapsed;
+  } else {
+    // leaf item
+    treeItem.collapsibleState = TreeItemCollapsibleState.None;
+  }
+
+  if (treeItem.collapsibleState !== origCollapsibleState) {
```

Comment:
Is this supposed to always ensure that the new id value changes?

Reply:
Yep, it just needs to change from the "without search" `id` being used, due to the odd behavior we've seen in the past when dynamically adjusting collapsible states and not changing the `id` like with the "Confluent Cloud" container between sign-in/sign-out:
https://github.com/confluentinc/vscode/blob/8591d90e00ad3955d48b3c53ff8d38349faab166/src/viewProviders/resources.ts#L430-L431

Reply:
It would seem that the id would keep growing, if, say, someone does a search starting from the search button, *then* does one through the command palate, etc.

Probably inconsequential, however.